CommunicationLayer/CommDeviceControl/UdpConnectionService.cpp

```diff
@@ -14,38 +26,83 @@ UdpConnectionService::~UdpConnectionService()
 
 void  UdpConnectionService::setMulticastNetwork(QHostAddress groupAddress, quint16 port)
 {
-    groupAddress_ = groupAddress;
-    port_ = port;
+    //groupAddress_ = groupAddress;
+    //port_ = port;
+
+
+
 }
 
-bool UdpConnectionService::connectToDataSource()
+void UdpConnectionService::setupChannel()
 {
-    if (groupAddress_.isNull() || port_ == 0)
-    {
-        emit connectionFailed("Group address and port number is not set!");
-        return false;
-    }
+    quint32 i = 0;
 
-    if (socket_.bind(QHostAddress::AnyIPv4, port_, QUdpSocket::ShareAddress) &&
-            socket_.joinMulticastGroup(groupAddress_))
-    {
-        emit connectionSucceeded();
-        return true;
-    }
-    else
+    do
```

Comment:
Don't use a while statement and block the whole program. Use QT's QTimer, combining with signals and slots to try a reconnection every so often.